### ISSUE TITLE
Make github compare link in `bun upgrade --canary` use the git sha of the new build instead of `main`

### ISSUE THREAD

**Issue by leeoniya** [oven-sh/bun#4115]
### What platform is your computer?

Linux 6.1.4-arch1-1 x86_64 unknown

### How did you attempt to install or upgrade?

![image](https://github.com/oven-sh/bun/assets/43234/6c540735-8a1b-470e-8265-77c082c44afb)

### What do you see instead?

one hash is from 30min ago, the other is from 3 days ago. they can't both be correct?

**Comment by Jarred-Sumner** [comment]
The previous hash is the one you were on before. The 2nd hash is the one you're on now.

**Comment by leeoniya** [comment]
but it won't always be `main` then since it needs to wait for CI? so `main` should actually be the revision hash?

can the compare url use shortened hashes, too?

**Comment by Jarred-Sumner** [comment]
Yeah, it’s hardcoded to main because we didn’t have —revision when this
code was last updated

**Comment by birkskyum** [comment]
Related to
- #3749

**Comment by birkskyum** [comment]
I might be more intuitive to always show the changelog between the newly installed canary release, and the release you'd have installed if the `--canary` flag wasn't provided.
